web/modules/custom/bglobal_angular/components/voices_spotlight/component.ts

```diff
@@ -0,0 +1,59 @@
+import {Component} from "@angular/core";
+import {ElementRef} from '@angular/core';
+import {Inject} from '@angular/core';
+import {Http} from '@angular/http';
+import 'rxjs/add/operator/map';
+
+@Component({
+  moduleId: __moduleName,
+  selector: "voices-spotlight",
+  templateUrl: 'template.html'
+})
+
+export class VoicesSpotlight{
+  public post: string = 'all';
+  public tag: string = 'all';
+  public program: string = 'all';
+  public baseUrl: string = 'https://voices.berkeley.edu/single-post/';
+  public teaser = {
+    content: "",
+    image: {
+      src: "https://voices.berkeley.edu/sites/default/files/styles/blog_well/public/global-logo-celebration-blog.jpg?itok=5PlyyzNQ",
+    }
+    link: "",
+    path: "",
+    tags: [],
+    title: "",
+  };
+
+  // Constructor method sets our data from the JSON callback.
+  constructor(
+    @Inject(Http) http: Http,
+    @Inject(ElementRef) elementRef: ElementRef) {
+
+    this.setVariables(elementRef);
+    var fetchUrl = this.baseUrl + this.post + '/tag/' + this.tag + '/cert/' + this.program + "/area/International";
+
+    // TODO: error handling.
```

Comment:
We really should fix the error handler now.

Reply:
Addressing the error handler in another pull request.